pkg/generic/thrift/write.go

```diff
@@ -208,6 +208,38 @@ func nextJSONWriter(data *gjson.Result, t *descriptor.TypeDescriptor, opt *write
 	return v, fn, nil
 }
 
+func getDefaultValueAndWriter(t *descriptor.TypeDescriptor, opt *writerOption) (interface{}, writer, error) {
```

Comment:
for multi cases switch, we have a more high performance coding style: 

```
func init() {
    var detaultValueWriter = map[Type]struct{interface{}, writer}{}
    detaultValueWriter[descriptor.BOOL] = {...}
}

func getDefaultValueAndWriter(t *descriptor.TypeDescriptor, opt *writerOption) (interface{}, writer, error) {
    return detaultValueWriter[t.Type].XXX. detaultValueWriter[t.Type].XXX, nil
}
```

https://hashrocket.com/blog/posts/switch-vs-map-which-is-the-better-way-to-branch-in-go